src/topography.f90

```diff
@@ -443,6 +444,29 @@ subroutine topography_min_max_depth(this, vgrid, level)
 
   end subroutine topography_min_max_depth
 
+  !-------------------------------------------------------------------------
+  subroutine topography_fill_fraction(this, sea_area_fraction)
+    class(topography_t), intent(inout) :: this
+    real(real32), intent(in) :: sea_area_fraction
+
+    write(output_unit,'(a,f7.2)') "Filling cells that have a sea area fraction smaller than ", sea_area_fraction
+
+    if (any(this%frac < sea_area_fraction)) then
+      where (this%frac < sea_area_fraction)
+        this%depth = 0.0
+        this%frac = 0.0
+      end where
+
+      if (this%lakes_removed == 'yes') then
+        ! We might have created new lakes, so rerun deseas
+        call this%deseas()
```

Comment:
I don't have strong feelings either way, but would it be more intuitive (ie maintain orthogonality between these operations) to omit `call this%deseas()` and instead set `lakes_removed = "no "` and add a warning in the help that lakes can be created so deseas should be re-run?

Reply:
I was unsure about this, also when I implemented a similar logic in `fix_nonadvective`). I went for the option that would prevent mistakes (like someone forgetting to run deseas after `fix_nonadvective` or `fill_fraction`), but maybe it's better to assume users know what they are doing and only do what exactly what they requested. It also helps understanding what is exactly happening if all the operations are explicitly listed in the script that creates and fixes the topography.

Reply:
I changed this. I will change the behavior of `fix_nonadvective` in a new PR, as I also want to include a few more changes.